### DISCUSSION TITLE
Subdomain routing

### DISCUSSION THREAD

**Discussion by tommerty** [withastro/roadmap#802] [Proposal]
### Body

# Summary

Add a way to handle subdomain routing

# Background & Motivation

So were building a platform for a link in bio and blog and stuff for users which is handled via [user].astro as a path, however we'd love a way to also include it as a subdomain like you can do with NextJS and middleware. This limitation has sadly sent us back to NextJS for this specific use case but would love to keep all our stack as Astro (our backend dashboard and landing is on Astro)

# Goals

A **concise, bulleted-list** outlining the intended goals of this RFC. 

- A way to map a path as a sub domain
- Would be helpful for othe SaaS builders such as ourselves

# Example

A good example is this project on Vercel: https://vercel.com/templates/next.js/platforms-starter-kit

**Comment by lerenos** [comment]
I'm trying to solve the same problem, and I think it is possible though middleware (similar to the vercel starter kit you mentioned). I'll post here if I can make it work.

**Comment by dotnize** [comment]
#681

**Reply by lerenos** [reply]
Thanks!

**Reply by jackwayneright** [reply]
@dotnize, following that proposal you mention there and it's subsequent proposal stages (https://github.com/withastro/roadmap/discussions/681, https://github.com/withastro/roadmap/issues/896, https://github.com/withastro/roadmap/pull/901) and feature PR (https://github.com/withastro/astro/pull/10853), it seems Astro ended up with the route rewrite capability (https://docs.astro.build/en/guides/routing/#rewrites). However, it's not immediately clear to me how this can be used to to use multiple subdomains within the same Astro project. Are you able to clarify this? In particular, it's not clear how to have the routing detect which subdomain is the current subdomain. Thank you for your time!

**Reply by jackwayneright** [reply]
Ah, I missed that the suggestion is using `rewrite` in combination with the [Middleware](https://docs.astro.build/en/guides/middleware/). While a guide would be useful (but perhaps niche) I think I now see how I could combine the two together to get it to handle subdomains properly.

**Reply by jsve** [reply]
@jackwayneright - did you find a way to put these concepts together in a working solution? I'm at a loss for how to do this. The docs seem to only handle path rewrites.
